prov/udp/src/udpx_init.c

```diff
@@ -88,7 +100,16 @@ static void udpx_getinfo_ifs(struct fi_info **info)
 	}
 	freeifaddrs(ifaddrs);
 
-	if (head) {
+	if (head || loopback) {
+		if(!head) { /* loopback interface only? */
+			head = loopback;
+		} else {
+			/* append loopback interfaces to tail */
+			assert(tail);
+			assert(!tail->next);
+			tail->next = loopback;
+		}
```

Comment:
maybe we should just form 2 lists, one for loopback, and another for non-loopback, then merge them at the end?  I can't think of a very clean way to handle this without duplicating a bunch of code based on loopback or not.